client/AI/AIWrapper.cpp

```diff
@@ -394,11 +394,6 @@ namespace {
             return 0;
         }
 
-        if (!empire->ProducibleItem(BuildType::BT_BUILDING, item_name, location_id)) {
-            ErrorLogger() << "IssueEnqueueBuildingProductionOrder : specified item_name and location_id that don't indicate an item that can be built at that location";
-            return 0;
-        }
-
```

Comment:
The human client allows it, I see no reason, why the AI should not be allowed to do it.

Reply:
Also consider whether the order will execute on the server... If not, is being able to issue an broken order useful?

Reply:
I got an error from when the AI issued an orbital dry dock right after the basic ship yard. As far as I know all cases where a building is enqueueable, but not producible are cases where a prerequisite is also in the queue.

Reply:
I suppose this should be a separate commit, if not a separate pull request, as it's seemingly unrelated to refactoring the getNamed* interface